### DISCUSSION TITLE
Feature idea: show build results for old releases

### DISCUSSION THREAD

**Discussion by kevinrenskers** [SwiftPackageIndex/SwiftPackageIndex-Server#1667] [Ideas]
Maybe I'm missing something, but it seems impossible to see build results of old releases? What if I want to look at or point to an older version of my package that is still available for older Swift versions, I'd love to be able to do that.

**Comment by finestructure** [comment]
We've had to set a limit to what build results we keep around because of storage and complexity. By tracking only the main branch as well as the latest release and the latest beta (if there is one), we can limit the number of builds we keep to ~300,000, whereas a complete history would be ~7,200,000.

**Reply by Sherlouk** [reply]
Is there a way to perhaps simplify the "archival" data so it's not so large and is simply a lightweight snapshot of the results as opposed to full detailed build logs like we currently support?

At most it's just an array of Swift versions and an array of platforms per version. Doesn't feel awfully complex and difficult but please educate me as I know this is a very advanced system and rooted in how SPI runs.

**Reply by kevinrenskers** [reply]
Yeah, I'm mostly interested in the checkmarks, not so much the build logs. Also shields for old versions would be awesome :)

That way you can include shields for specific versions in your readme.

**Reply by Sherlouk** [reply]
Yeah I love the idea of shields for each version as you can then specify what tags to use in the README next to confirmed compatibility 😀

I think it also plays into the package maturity if it's been receiving updates to Swift versions etc, but also how fast they drop support etc.
